healpy/newvisufunc.py

```diff
@@ -150,45 +237,82 @@ def mollview(
             grid_map = m[grid_pix]
 
             # plot
-            ret = plt.pcolormesh(
-                longitude,
-                latitude,
-                grid_map,
-                vmin=min,
-                vmax=max,
-                rasterized=True,
-                **kwargs
-            )
-
+            if return_only_data: # exit here when dumping the data
+                return [longitude,  latitude, grid_map]
+            if projection_type is not '3d': # test for 3d plot
+                ret = plt.pcolormesh(
+                    longitude,
+                    latitude,
+                    grid_map,
+                    vmin=min,
+                    vmax=max,
+                    rasterized=True,
+                    cmap=cmap,
+                    shading='auto',
+                    **kwargs
+                )
+            elif projection_type is '3d': # test for 3d plot
+                LONGITUDE, LATITUDE = np.meshgrid(longitude, latitude)
+                ret = ax.plot_surface(LONGITUDE, LATITUDE, grid_map, cmap=cmap, vmin=min,  vmax=max, rasterized=True, **kwargs)
         # graticule
         plt.grid(graticule)
+        
         if graticule:
-            longitude_grid_spacing = 60  # deg
-            ax.set_longitude_grid(longitude_grid_spacing)
-            if width < 10:
-                ax.set_latitude_grid(45)
+            if projection_type == "aitoff" or projection_type ==  "hammer" or  projection_type == "lambert" or projection_type == "mollweide":
```

Comment:
`projection_type in ["hammer", ...]`, or better save list in variable to reuse later.

Reply:
Now, these projections are grouped to `geographic_projections`.